web/web/src/app/metalakes/metalake/rightContent/CreateTableDialog.js

```diff
@@ -0,0 +1,643 @@
+  columns: yup.array().of(
+    yup.object().shape({
+      name: yup.string().required(),
+      type: yup.string().required(),
+      nullable: yup.boolean(),
+      comment: yup.string()
+    })
+  ),
+  propItems: yup.array().of(
+    yup.object().shape({
+      required: yup.boolean(),
+      key: yup.string().required(),
+      value: yup.string().when('required', {
+        is: true,
+        then: schema => schema.required()
+      })
+    })
+  )
+})
+
+// Dialog transition component
+const Transition = forwardRef(function Transition(props, ref) {
+  return <Fade ref={ref} {...props} />
+})
+
+/**
+ * Main CreateTableDialog component
+ * Handles creation and editing of tables with columns and properties
+ */
+const CreateTableDialog = props => {
+  // Destructure props
+  const { open, setOpen, type = 'create', data = {} } = props
+
+  // Get URL parameters
+  const searchParams = useSearchParams()
+  const metalake = searchParams.get('metalake')
+  const catalog = searchParams.get('catalog')
+  const catalogType = searchParams.get('type')
+  const schemaName = searchParams.get('schema')
+
+  // Component state
+  const [innerProps, setInnerProps] = useState([])
+  const [tableColumns, setTableColumns] = useState([])
+  const [initialTableData, setInitialTableData] = useState()
+  const dispatch = useAppDispatch()
+
+  // Initialize form with react-hook-form
+  const {
+    control,
+    reset,
+    setValue,
+    getValues,
+    handleSubmit,
+    trigger,
+    formState: { errors }
+  } = useForm({
+    defaultValues: defaultFormValues,
+    mode: 'all',
+    resolver: yupResolver(schema)
+  })
+
+  /**
+   * Handle changes to property form fields
+   * Validates keys and checks for duplicates
+   */
+  const handlePropertyChange = ({ index, event }) => {
+    let updatedProps = [...innerProps]
+    updatedProps[index][event.target.name] = event.target.value
+
+    if (event.target.name === 'key') {
+      const isInvalidKey = !keyRegex.test(event.target.value)
+      updatedProps[index].invalid = isInvalidKey
+    }
+
+    const nonEmptyKeys = updatedProps.filter(item => item.key.trim() !== '')
+    const groupedKeys = groupBy(nonEmptyKeys, 'key')
+    const hasDuplicateKeys = Object.keys(groupedKeys).some(key => groupedKeys[key].length > 1)
+
+    if (hasDuplicateKeys) {
+      updatedProps[index].hasDuplicateKey = hasDuplicateKeys
+    } else {
+      updatedProps.forEach(item => (item.hasDuplicateKey = false))
+    }
+
+    setInnerProps(updatedProps)
+    setValue('propItems', updatedProps)
+  }
+
+  /**
+   * Handle changes to column fields
+   */
+  const handleColumnChange = ({ index, field, value }) => {
+    let updatedColumns = [...tableColumns]
+    updatedColumns[index][field] = value
+    setTableColumns(updatedColumns)
+    setValue('columns', updatedColumns)
+  }
+
+  /**
+   * Add a new empty column
+   */
+  const addColumn = () => {
+    const newColumn = { name: '', type: '', nullable: false, comment: '' }
```

Comment:
nullable default value is 'true', but depending on the needs of the product, this field is used with `Not null`, and the default value of `Not null` will be `false`

Reply:
Should I set default value depending on column type?

Reply:
I set the default value to true to match the API behavior.
![image](https://github.com/user-attachments/assets/60c5a759-956f-49fd-b6fe-dbc66860ae6c)